lib/text_diff.ml

```diff
@@ -206,20 +234,34 @@ and pair_of_manifest_diff manifest_diff =
   | Removed te -> (Some te, None)
   | Modified { reference; current } -> (Some reference, Some current)
 
-and type_header_to_line name private_flag type_expr_opt abstract =
+and type_header_to_line name private_flag type_expr_opt type_params abstract =
   let private_flag_str = string_of_private_flag private_flag in
   let manifest_str = string_of_manifest type_expr_opt in
+  let type_params_str = string_of_type_params type_params in
   let equal_sign =
     if abstract && private_flag_str = "" && manifest_str = "" then "" else " ="
   in
-  Printf.sprintf "type %s%s%s%s" name equal_sign private_flag_str manifest_str
+  Printf.sprintf "type %s%s%s%s%s" type_params_str name equal_sign
+    private_flag_str manifest_str
 
 and string_of_private_flag private_flag =
   match private_flag with Public -> "" | Private -> " private"
 
 and string_of_manifest manifest =
   match manifest with None -> "" | Some te -> " " ^ typ_expr_to_line te
 
+and string_of_type_params (type_params : Types.type_expr list) =
+  let wrap = if List.length type_params > 1 then true else false in
```

Comment:
Really nitpicking but `if ... then true else false` can always be simpilified!
```suggestion
  let wrap = List.length type_params > 1 in
```
Nitpicking even more, I would do it like that to avoid computing the length:
```suggestion
  let wrap = match type_params with _ :: _ :: _ -> true | _ -> false in
```